src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    if (this.length > 0) {
+      for (let i = 0; i < this.length; i++) {
+        if (this[i] === null || this[i] === undefined) {
+          result += '';
+        } else {
+          result += this[i];
+        }
```

Comment:
Я спочатку теж так написав а виявляється можна коротше)
```suggestion
        if (this[i] !== null && this[i] !== undefined) {
          newName += this[i];
        }      
```